network/irc/irc.go

```diff
@@ -596,6 +611,23 @@ func (bot *ircBot) act(theLine *line.Line) {
 		return
 	}
 
+	isAskingForSasl := theLine.User == "" && strings.HasSuffix(theLine.Raw, " CAP * ACK :sasl")
```

Comment:
The raw input may slightly vary across ircds; better compare the parsed command & parameters.